keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Resource.kt

```diff
@@ -63,15 +62,11 @@ data class Resource<T : ResourceSpec>(
 /**
  * External representation of a resource that would be submitted to the API
  */
+@JsonDeserialize(using = SubmittedResourceDeserializer::class)
 data class SubmittedResource<T : ResourceSpec>(
   val metadata: Map<String, Any?>,
   val apiVersion: ApiVersion,
   val kind: String,
-  @JsonTypeInfo(
-    use = Id.NAME,
-    include = As.EXTERNAL_PROPERTY,
-    property = "kind"
-  )
   val spec: T
```

Comment:
Now use a custom deserializer instead of trying to use `@JsonTypeInfo`. There's no way to get that to use a compound properly without it also appearing as a single value in the JSON.